### DISCUSSION TITLE
cubeb?

### DISCUSSION THREAD

**Discussion by jacksongoode** [jpochyla/psst#318] [Q&A] [answered]
Hi, just saw that you added cubeb as a backend and I was wondering if you plan on migrating to it and what the benefits might be in your view! Glad to see some more work on this app!

**Comment by jpochyla** [comment, marked as answer]
Hi! `cubeb` behaves a bit better on MacOS, so I've planned to integrate it for some time already. I.e. it correctly routes the playback to/from heaphone jack and the internal laptop speakers. It also supports notifications on device change, so we can i.e. pause the playback when the headphone jack is disconnected (but continue playback when transferring from the internal speaker). It also opens up the possibility of integrating more backend crates -- I've given up on having a single, Rust-only, audio output library.
